### PR TITLE
graceful exceptions

### PR DESCRIPTION
**Pull request by iRebbok**
Another bulky non-tested PR...

I know many of us (probably everyone except @sanyae2439 😅) don't fuck up with mono & dnSpy to make debugging a dll easier.
This PR doesn't include a debugger but makes all possible exceptions graceful (see https://github.com/benaadams/Ben.Demystifier#make-error-logs-more-productive).

Not sure if this will help anyone except server owners with erroring plugins by reducing the size of the logs (probably).

### CODE REVIEW

**Comment by Michal78900** [comment]
I think I had similar idea but you were faster and probably make it better :trollface:

**Comment by iRebbok** [comment]
> I think I had similar idea but you were faster and probably make it better :trollface:

I don't think I'm as fast as you in fixing build errors...

**Comment by iopietro** [comment]
Good addition, I like it 👍.

File: Exiled.Events/Patches/Events/Player/ChangingRole.cs [hunk 1]
```diff
@@ -166,7 +167,7 @@ private static bool Prefix(CharacterClassManager __instance, ref RoleType classi
             }
             catch (Exception e)
             {
-                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.ChangingRole: {e}\n{e.StackTrace}");
+                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.ChangingRole: {e}\n{e.ToStringDemystified()}");
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/ChangingRole.cs:170]:
```suggestion
                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.ChangingRole:\n{e.ToStringDemystified()}");
```

File: Exiled.Events/Patches/Events/Player/ChangingItem.cs [hunk 2]
```diff
@@ -56,7 +57,7 @@ private static void Prefix(Inventory __instance, int i)
             }
             catch (Exception e)
             {
-                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.ChangingItem: {e}\n{e.StackTrace}");
+                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.ChangingItem: {e}\n{e.ToStringDemystified()}");
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/ChangingItem.cs:60]:
```suggestion
                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.ChangingItem:\n{e.ToStringDemystified()}");
```

File: Exiled.Events/Patches/Events/Player/ChangingGroup.cs [hunk 3]
```diff
@@ -34,7 +35,7 @@ private static bool Prefix(ServerRoles __instance, UserGroup group)
             }
             catch (Exception e)
             {
-                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.ChangingGrounp: {e}\n{e.StackTrace}");
+                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.ChangingGrounp: {e}\n{e.ToStringDemystified()}");
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/ChangingGroup.cs:38]:
```suggestion
                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.ChangingGrounp:\n{e.ToStringDemystified()}");
```

File: Exiled.Events/Patches/Events/Player/BanningAndKicking.cs [hunk 4]
```diff
@@ -158,7 +159,7 @@ private static bool Prefix(GameObject user, int duration, string reason, string
             }
             catch (Exception e)
             {
-                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.BanningAndKicking: {e}\n{e.StackTrace}");
+                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.BanningAndKicking: {e}\n{e.ToStringDemystified()}");
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/BanningAndKicking.cs:162]:
```suggestion
                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.BanningAndKicking:\n{e.ToStringDemystified()}");
```

File: Exiled.Events/Patches/Events/Player/EnteringFemurBreaker.cs [hunk 5]
```diff
@@ -61,7 +62,7 @@ private static bool Prefix(CharacterClassManager __instance)
             }
             catch (Exception e)
             {
-                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.EnteringFemurBreaker: {e}\n{e.StackTrace}");
+                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.EnteringFemurBreaker: {e}\n{e.ToStringDemystified()}");
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/EnteringFemurBreaker.cs:65]:
```suggestion
                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.EnteringFemurBreaker:\n{e.ToStringDemystified()}");
```

File: Exiled.Events/Patches/Events/Player/Hurting.cs [hunk 6]
```diff
@@ -73,7 +74,7 @@ private static bool Prefix(PlayerStats __instance, ref PlayerStats.HitInfo info,
             }
             catch (Exception e)
             {
-                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.Hurting: {e}\n{e.StackTrace}");
+                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.Hurting: {e}\n{e.ToStringDemystified()}");
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/Hurting.cs:77]:
```suggestion
                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Player.Hurting:\n{e.ToStringDemystified()}");
```

File: Exiled.Loader/Loader.cs [hunk 7]
```diff
@@ -330,7 +350,8 @@ public static void DisablePlugins()
                 }
                 catch (Exception exception)
                 {
-                    Log.Error($"Plugin \"{plugin.Name}\" threw an exception while disabling: {exception}");
+                    Log.Error(
```
Comment (iopietro) [inline Exiled.Loader/Loader.cs:353]: Do we need a new line here? (And in every `Log.Error(...)` above)

**Review by iopietro — APPROVED** [review]
Everything else is fine.

File: Exiled.Loader/Loader.cs (hunk 7 above)
Comment (iRebbok) [reply Exiled.Loader/Loader.cs:353]: Not really, just Rider's code cleanup, rolling back that mess.

**Review by BuildBoy12 — APPROVED** [review]

**Pull request merged**